### PR TITLE
Support for custom S3 server (eq. DigitalOcean Spaces) incl. ACL support

### PR DESCRIPTION
**Pull request by stvno**
Building upon #364 to support ACL when uploading tiles. Digital Ocean's default policy is to set the files private and that makes it difficult to use.
This PR adds ACL support to @mikkokut work

### CODE REVIEW

File: mapproxy/config/loader.py [hunk 1]
```diff
@@ -1121,6 +1130,9 @@ def _s3_cache(self, grid_conf, file_ext):
             directory_layout=directory_layout,
             bucket_name=bucket_name,
             profile_name=profile_name,
+            region_name=region_name,
+            endpoint_url=endpoint_url,
+	    access_control_list=access_control_list
```
Comment (olt) [inline mapproxy/config/loader.py]: There is a tab character.

File: doc/caches.rst [hunk 2]
```diff
@@ -434,7 +434,7 @@ You can set the ``sources`` to an empty list, if you use an existing geopackage
 
 .. versionadded:: 1.10.0
```
Comment (olt) [inline doc/caches.rst]: .. versionadded:: 1.11.0
      ``region_name``, ``endpoint_url`` and ``access_control_list``

File: doc/caches.rst [hunk 3]
```diff
@@ -454,6 +454,15 @@ Available options:
...
   You can set the default profile with ``globals.cache.s3.profile_name``.
 
+``region_name``:
+  Optional name of the region. You can set the default region_name with ``globals.cache.s3.region_name``
+
+``endpoint_url``:
+  Optional endpoint_url for the S3. You can set the default endpoint_url with ``globals.cache.s3.endpoint_url``.
+
+``access_control_list``:
+  Optional access control list for the S3. You can set the default access_control_list with ``globals.cache.s3.access_control_list``.
```
Comment (olt) [inline doc/caches.rst]: Is this DO specific, or does this also work for S3? Please add a comment if it is DO specific.

Comment (stvno) [reply doc/caches.rst]: This is a generic S3 API option, so it works on AWS, DO and any S3-like service

File: mapproxy/config/loader.py (hunk 1 above)
Comment (stvno) [reply mapproxy/config/loader.py]: grmbl stupid editors mumble mumble

**Comment by stvno** [comment]
removed the tab & added the extra version info

**Comment by olt** [comment]
Hmmm, tests are still failing for Python 3 because of some indention issues (https://travis-ci.org/mapproxy/mapproxy/jobs/389591246#L2922)
The Travis tests should pass, you can ignore the appveyor tests.

**Comment by stvno** [comment]
I've opened the files in a proper editor and AFAIK there are no tabs left. I also updated the documentation with an example for access_control_list

**Comment by mikkokut** [comment]
@stvno  I think I can close my pull request as this has also my changes? Should the description be updated to cover also the original feature?

**Comment by stvno** [comment]
@mikkokut yeah I forked your code, so it is all in there, thanks for doing the hard bits :)

**Comment by olt** [comment]
Merged. Thanks @stvno and @mikkokut

**Pull request merged**